docs/API.md

```diff
@@ -0,0 +1,230 @@
+> ```
+> or using the default import and accessing the component using `PaletteProvider` property:
+> ```javascript
+> import MaterialPalette from 'react-native-material-palette';
+> // MaterialPalette.PaletteProvider
+> ```
+
+### Description
+`MaterialPaletteProvider` is a component, which handles palette creation and provides the access to the palette instance for _connected_ components (via `withMaterialPalette`) using context. Ideally, `MaterialPaletteProvider` should be placed at the top of components tree, so that all nested components can _connect_ to it.
+
+The concept is very similar to `Provider` component from `react-redux`.
+
+### Props
+* `image: Image` (__required__) - Local image to create palette from (`require('path/to/image')`) or object with remote URI adress from which the image can be downloaded (`{ uri: 'http://some-domain.ext/image.png' }`) - same as `image` in `MaterialPalette.create` function.
+
+* `options?: Options` (optional) - Options for palette creation - same as `options` in `MaterialPalette.create` function:
+  ```javascript
+  type Options = {
+    region?: { top: number, left: number, bottom: number, right: number },
+    maximumColorCount?: number = 16,
+    type?: ColorProfile = 'vibrant',
+    types?: Array<ColorProfile> = []
+  }
+  ```
+
+* `defaults?: PaletteDefaults` (optional) - Global defaults which will be propagated to each _connected_ component, alongside with palette instance, which will be used, if the specific color profile is not found:
+  ```javascript
+  type ColorProfile =
+    | 'muted'
+    | 'vibrant'
+    | 'darkMuted'
+    | 'darkVibrant'
+    | 'lightMuted'
+    | 'lightVibrant';
+  
+  type DefaultSwatch = {
+    color: string,
+    bodyTextColor: string,
+    titleTextColor: string,
+  };
+
+  type PaletteDefaults = {
+    [key: ColorProfile]: DefaultSwatch,
+  };
+  ```
+
+* `waitForPalette?: boolean | React$Component<*, *, *> | ((...args: *) => React$Element<*>)` (optional) - If specified, while waiting for palette to be created, will render either `null` (if `waitForPalette === true`) or the passed component:
+  * `<MateriaPaletteProvider waitForPalette>` - will render `null`,
+  * `<MateriaPaletteProvider waitForPalette={true}>` - will render `null`,
+  * `<MateriaPaletteProvider waitForPalette={SpinnerComponent}>` - will render `SpinnerComponent`,
+  * `<MateriaPaletteProvider waitForPalette={() => <Text>Loading</Text>)}>` - will render `Text` component with _Loading_.
+
+* `onError?: (error: Error) => void` (optional) - Error handler, called if the palette failed to create.
+
+* `onInit?: () => void` - (optional) - Init handler, called when the `MaterialPaletteProvider` is just about to start creating the palette.
+
+* `onFinish?: (palette: PaletteInstance, globalDefaults: PaletteDefaults) => void` - (optional) - Finish handler, called when the palette is created, but before it gets propagated to _connected_ components - use it, if you want to mutate the palette instance.
```

Comment:
If for some reason, user want to mutate palette, `onFinish` should be used for that. Besides it's an escape hatch for case if someone want to render splash screen and then animate the transision.